chord_metadata_service/patients/api_views.py

```diff
@@ -213,3 +213,72 @@ def get(self, request, *args, **kwargs):
                 "experiment_type": experiment_types
             }
         })
+
+
+class BeaconListIndividuals(APIView):
+    """
+    View to return lists of individuals filtered using search terms from katsu's config.json.
+    Uncensored equivalent of PublicListIndividuals.
+    """
+    def filter_queryset(self, queryset):
+        # Check query parameters validity
+        qp = self.request.query_params
+        if len(qp) > settings.CONFIG_PUBLIC["rules"]["max_query_parameters"]:
+            raise ValidationError(f"Wrong number of fields: {len(qp)}")
+
+        search_conf = settings.CONFIG_PUBLIC["search"]
+        field_conf = settings.CONFIG_PUBLIC["fields"]
+        queryable_fields = {
+            f"{f}": field_conf[f] for section in search_conf for f in section["fields"]
+        }
+
+        for field, value in qp.items():
+            if field not in queryable_fields:
+                raise ValidationError(f"Unsupported field used in query: {field}")
+
+            field_props = queryable_fields[field]
+            options = get_field_options(field_props)
+            if value not in options \
+                    and not (
+                        # case-insensitive search on categories
+                        field_props["datatype"] == "string"
+                        and value.lower() in [o.lower() for o in options]
+                    ) \
+                    and not (
+                        # no restriction when enum is not set for categories
+                        field_props["datatype"] == "string"
+                        and field_props["config"]["enum"] is None
+                    ):
+                raise ValidationError(f"Invalid value used in query: {value}")
+
+            # recursion
+            queryset = filter_queryset_field_value(queryset, field_props, value)
+
+        return queryset
+
+    def get(self, request, *args, **kwargs):
+        if not settings.CONFIG_PUBLIC:
+            return Response(settings.NO_PUBLIC_DATA_AVAILABLE)
```

Comment:
should this not be a 404 instead of a 200? fine to leave as-is for consistency with bento public responses but generally i think in API design it's good to put as much meaning in the status code as possible

Reply:
Yeah, this was a straight copypaste of the bento_public code, and it struck me as odd also. I won't break the old code, but I'll change the beacon responses to 400.

Reply:
hmm i think of 400s more meaning something is wrong with the request... if the client request is good, I still think this should be a 404 (or maybe a 500 if we treat it as a configuration error)

Reply:
Ah, okay, I actually re-read the documentation for 404 and this sounds reasonable ("missing resource"). 204 (No content) is also possible but probably a little obscure. I'll change it to 404.